components/director/internal/repo/list_test.go

```diff
@@ -0,0 +1,79 @@
+package repo_test
+
+import (
+	"context"
+	"database/sql/driver"
+	"regexp"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
+	"github.com/kyma-incubator/compass/components/director/internal/repo/testdb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestList(t *testing.T) {
+	givenTenant := uuidB()
+	peterID := uuidA()
+	homerID := uuidC()
+	peter := User{FirstName: "Peter", LastName: "Griffin", Age: 40, Tenant: givenTenant, ID: peterID}
+	peterRow := []driver.Value{peterID, givenTenant, "Peter", "Griffin", 40}
+	homer := User{FirstName: "Homer", LastName: "Simpson", Age: 55, Tenant: givenTenant, ID: homerID}
+	homerRow := []driver.Value{homerID, givenTenant, "Homer", "Simpson", 55}
+
+	sut := repo.NewLister("users", "tenant_col",
+		[]string{"id_col", "tenant_col", "first_name", "last_name", "age"})
+
+	t.Run("lists all items successfully", func(t *testing.T) {
+		db, mock := testdb.MockDatabase(t)
+		defer mock.AssertExpectations(t)
+
+		rows := sqlmock.NewRows([]string{"id_col", "tenant_col", "first_name", "last_name", "age"}).
+			AddRow(peterRow...).
+			AddRow(homerRow...)
+		mock.ExpectQuery(regexp.QuoteMeta(`SELECT id_col, tenant_col, first_name, last_name, age FROM users WHERE tenant_col=$1`)).WithArgs(givenTenant).WillReturnRows(rows)
+		ctx := persistence.SaveToContext(context.TODO(), db)
+		var dest UserCollection
+
+		err := sut.List(ctx, givenTenant, &dest)
+		require.NoError(t, err)
+		assert.Len(t, dest, 2)
+		assert.Equal(t, peter, dest[0])
```

Comment:
Is the order always the same?

Reply:
With mocked DB - yes, but I'll change those asserts to `Contains` to make the example better.